modules/jwplayerBidAdapter.js

```diff
@@ -50,4 +90,110 @@ export const spec = {
   // onBidderError: function({ error, bidderRequest }) {}
 };
 
+function buildRequest(bidRequest, bidderRequest) {
+  const openrtbRequest = {
+    id: bidRequest.bidId,
+    imp: buildRequestImpression(bidRequest, bidderRequest),
+    site: buildRequestSite(bidRequest),
+    device: buildRequestDevice()
+  };
+
+  // Attaching GDPR Consent Params
+  if (bidderRequest.gdprConsent) {
+    deepSetValue(openrtbRequest, 'user.ext.consent', bidderRequest.gdprConsent.consentString);
+    deepSetValue(openrtbRequest, 'regs.ext.gdpr', (bidderRequest.gdprConsent.gdprApplies ? 1 : 0));
+  }
+
+  // CCPA
+  if (bidderRequest.uspConsent) {
+    deepSetValue(openrtbRequest, 'regs.ext.us_privacy', bidderRequest.uspConsent);
+  }
+
+  return JSON.stringify(openrtbRequest);
+}
+
+function buildRequestImpression(bidRequest) {
+  const impressionObject = {
+    id: bidRequest.adUnitCode,
+  };
+
+  impressionObject.video = buildImpressionVideo(bidRequest);
+
+  const bidFloorData = buildBidFloorData(bidRequest);
+  if (bidFloorData) {
+    impressionObject.bidfloor = bidFloorData.floor;
+    impressionObject.bidfloorcur = bidFloorData.currency;
+  }
+
+  impressionObject.ext = buildImpressionExtension(bidRequest);
+
+  return [impressionObject];
+}
+
+function buildImpressionVideo(bidRequest) {
+  const videoParams = deepAccess(bidRequest, 'mediaTypes.video', {});
+
+  const video = {};
+
+  VIDEO_ORTB_PARAMS.forEach((param) => {
+    if (videoParams.hasOwnProperty(param)) {
+      video[param] = videoParams[param];
+    }
+  });
+
+  return video;
+}
+
+function buildImpressionExtension(bidRequest) {
+  return {
+    appnexus: {
+      placement_id: bidRequest.params.placementId
+    }
+  };
+}
+
+function buildBidFloorData(bidRequest) {
+  const {params} = bidRequest;
+  const currency = params.currency || 'USD';
+
+  let floorData;
+  if (isFn(bidRequest.getFloor)) {
+    const bidFloorRequest = {
+      currency: currency,
+      mediaType: VIDEO,
+      size: '*'
+    };
+    floorData = bidRequest.getFloor(bidFloorRequest);
+  } else if (params.bidfloor) {
+    floorData = {floor: params.bidfloor, currency: currency};
+  }
+
+  return floorData;
+}
+
+function buildRequestSite(bidRequest) {
+  const configSite = config.getConfig('ortb2.site');
+
+  const site = {
+    domain: (configSite && configSite.domain) ? configSite.domain : window.location.hostname,
```

Comment:
this is good, but in Player Bidding we are setting these properties in a different place. I am not sure why there are 2 places for the same information, but these are properties that exist in the config, so it is possible that they are used in other places.
In player bidding we set to `config.publisherDomain` and `config.pageUrl`. I think ideally those would supersede.
So I recommend
```
const site = {
            domain: config.publisherDomain || configSite.domain || window.location.hostname,
            page: config.pageUrl || configSite.page || window.location.href
};
```

Reply:
fixed.